### PR TITLE
IPROTO-75 Conversion from JSON to Protobuf fails when using required non-scalar fields

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.jboss.org/browse/IPROTO-75

### CODE REVIEW

File: core/src/test/java/org/infinispan/protostream/domain/Account.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
    private String description;
    private Date creationDate;
    private Limits limits;
+   private Limits hard_limits;
```
Comment (anistor) [inline core/src/test/java/org/infinispan/protostream/domain/Account.java]: I'd use camel casing instead of ```hard_limits```, here, in toString, etc

**Comment by anistor** [comment]
LGTM. I'll backport this to 4.2.x also

File: core/src/test/java/org/infinispan/protostream/domain/Account.java (hunk 1 above)
Comment (gustavonalle) [reply core/src/test/java/org/infinispan/protostream/domain/Account.java]: fixed

**Comment by anistor** [comment]
Integrated in master and 4.2.x. Thanks @gustavonalle !

**Pull request merged**